addons/l10n_latam_invoice_document/models/account_move.py

```diff
@@ -41,8 +41,31 @@ def _auto_init(self):
         # for a Chilian or Argentian company (`res.company`) before installing `l10n_cl` or `l10n_ar`.
         if not column_exists(self.env.cr, "account_move", "l10n_latam_document_type_id"):
             create_column(self.env.cr, "account_move", "l10n_latam_document_type_id", "int4")
+
+        # Update the generic unique name constraint to not consider the purchases in latam companies.
+        # The name should be unique by partner for those documents.
+        if not index_exists(self.env.cr, "account_move_unique_name_latam"):
+            drop_constraint(self.env.cr, self._table, "account_move_unique_name")
         return super()._auto_init()
 
+    _sql_constraints = [(
+        'unique_name',  # rewrite the constraint defined in `account` after removing it in `_auto_init`
```

Comment:
(Also, why 2 constraints instead of adding the condition in it?

Reply:
> Or maybe it never happens and I just overthink 😁

It should indeed not happen. If it does, we'll implement something like that. But useless/too complicated for now.

> Also, why 2 constraints instead of adding the condition in it?

What do you mean "adding the condition"? You can't just change the constraint/index by amending it.
And the constraints are different: one is unique per journal and the other one per partner, each with a different domain.

Reply:
You're right